jax/_src/numpy/lax_numpy.py

```diff
@@ -127,8 +127,35 @@ def canonicalize_shape(shape: Any, context: str="") -> core.Shape:
 printoptions = np.printoptions
 set_printoptions = np.set_printoptions
 
-@util.implements(np.iscomplexobj)
 def iscomplexobj(x: Any) -> bool:
+  """Checks if the input is a complex number or an array containing complex elements.
+
+  JAX implementation of :func:`numpy.iscomplexobj`.
+
+  The function evaluates based on input type rather than value. Inputs with zero imaginary parts are still considered complex.
```

Comment:
We're not very strict about this, but we typically want to wrap to ~80 character lines.